### PR TITLE
update more 0.7 deprecations

### LINKED ISSUE #98: Tackle last remaining deprecation warnings
I've been unable to figure out what's causing the last remaining warnings.

https://travis-ci.org/JuliaML/LossFunctions.jl/jobs/411444160

There's some stuff hidden inside functions or an eval that make it hard to find the source of the problem.  I thought I'd open an issue in case someone thinks they can figure it out.  I probably won't make another attempt until after JuliaCon.

### PR DESCRIPTION
**Pull request by Evizero**
fixes #98

### CODE REVIEW

**Comment by Evizero** [comment]
I also sketched out the project file, but it doesn't quite work yet because dependencies don't have one yet (I think). Anyway I am checking it in and we can rename it later

**Comment by Evizero** [comment]
Seems the CI version for 0.7 is a bit behind. As long as it compiles on nightly it should be fine

**Comment by Evizero** [comment]
- The decreased coverage is spurious (the code it says isn't executed is literally shown in the logs to be executed)

- The remaining warnings about `precompile` are not from LossFunctions but from dependencies.

File: src/supervised/distance.jl [hunk 1]
```diff
@@ -112,8 +112,8 @@ It is strictly convex.
 const L2DistLoss = LPDistLoss{2}
 
 value(loss::L2DistLoss, difference::Number) = abs2(difference)
-deriv(loss::L2DistLoss, difference::T) where {T<:Number} = T(2) * difference
-deriv2(loss::L2DistLoss, difference::T) where {T<:Number} = T(2)
+deriv(loss::L2DistLoss, difference::T) where {T<:Number} = convert(T,2) * difference
+deriv2(loss::L2DistLoss, difference::T) where {T<:Number} = convert(T,2)
```
Comment (Evizero) [inline src/supervised/distance.jl]: the changes from `T(x)` to `convert(T,x)` seem unfortunate, but that seems to be the right thing to do here.

Comment (joshday) [reply src/supervised/distance.jl]: Agreed.  Ah, that must be what all the dual number warnings were about...

**Comment by joshday** [comment]
My understanding is that there will be some automated process that will add project files to all Julia repos, so we may not have to worry about that ourselves.

**Comment by Evizero** [comment]
Right. The reason I tried anyway was because initially i got weird warnings after I added `using Markdown`, even though its a stdlib which AFAIK we should not put into REQUIRE. 

```
┌ Warning: Package LossFunctions does not have Markdown in its dependencies:
│ - If you have LossFunctions checked out for development and have
│   added Markdown as a dependency but haven't updated your primary
│   environment's manifest file, try `Pkg.resolve()`.
│ - Otherwise you may need to report an issue with LossFunctions
└ Loading Markdown into LossFunctions from project dependency, future warnings for LossFunctions are suppressed.
```

**Pull request merged**